ApiDoctor.Console/Program.cs

```diff
@@ -2562,6 +2557,210 @@ private static HttpRequest PreProcessUrlForSnippetGeneration(HttpRequest request
+                            var permissionsDirectory = Path.Combine(Directory.GetParent(Path.GetDirectoryName(docFile.FullPath)).FullName, permissionsFileRelativePath);
+                            Directory.CreateDirectory(permissionsDirectory); // make sure the '/includes/permissions' directory exists
+                            var permissionsMarkdownFilePath = Path.Combine(permissionsDirectory, permissionsFileName);
+                            FancyConsole.WriteLine(FancyConsole.ConsoleSuccessColor, $"Writing permissions table to {permissionsMarkdownFilePath}");
+                            File.WriteAllText(permissionsMarkdownFilePath, permissionFileContents);
+
+                            // insert permissions block text into doc file
+                            var permissionsBlockText = $"<!-- {{ \"blockType\": \"permissions\", \"name\": \"{docFileName.Replace("-", "_")}\" }} -->\r\n" +
+                                $"[!INCLUDE [permissions-table](../{ReplaceWindowsByLinuxPathSeparators(Path.Combine(permissionsFileRelativePath, permissionsFileName))})]";
+                            IEnumerable<string> updatedFileContents = originalFileContents;
+                            updatedFileContents = updatedFileContents.Splice(insertionStartLine, insertionEndLine + 1 - insertionStartLine);
+                            updatedFileContents = FileSplicer(updatedFileContents.ToArray(), insertionStartLine - 1, permissionsBlockText);
+                            File.WriteAllLines(docFile.FullPath, updatedFileContents);
+                            parseStatus = PermissionsInsertionState.FindNextPermissionBlock;
+                            break;
+                        case PermissionsInsertionState.FindNextPermissionBlock:
+                            isBootstrapped = false;
+                            foundHttpRequestBlocks = 0;
+                            currentIndex = insertionStartLine + 2;
+                            originalFileContents = File.ReadAllLines(docFile.FullPath);
+                            parseStatus = PermissionsInsertionState.FindInsertionStartLine;
+                            insertionStartLine = insertionEndLine = httpRequestStartLine = httpRequestEndLine = -1;
+                            break;
+                        default:
+                            break;
+                    }
+                }
+            }
+            return true;
+        }
+
+        private static string GetPermissionsMarkdownTableForHttpRequestBlock(List<string> httpRequests, PermissionsDocument permissionsDoc)
+        {
+            // use the first HTTP Request, we are assuming the group of URLs will have the same set of permissions
+            var request = httpRequests.Where(x => !string.IsNullOrWhiteSpace(x)).FirstOrDefault();
+            if (request == null)
+                return null;
+
+            HttpParser.TryParseHttpRequest(request, out var parsedRequest);
```

Comment:
We should check if this returns true and then proceed. otherwise we'll get null reference errors trying to dereference `parsedRequest` if the parsing fails.